### DISCUSSION TITLE
How to enable "reactive mode"?

### DISCUSSION THREAD

**Discussion by psionic12** [emilk/egui#2937] [Q&A] [answered]
I found the concept "reactive mode" a lot in documents, but I searched for a while and can't find where to enable it, or how to implement it when integrate with backend renderers.

It looks to me that reactive mode can be controlled by setting `ControlFlow` by egui-winit, but I didn't find any docs to prove this. 

Are there any tutorials on how to do this? I used egui-winit with my own renderer.

**Comment by Wcubed** [comment, marked as answer]
As far as I can see, "reactive mode" is the default for egui now. It might have been different in the past, given that the readme on github confusingly mentions "You can also set up egui to only repaint when there is interaction (e.g. mouse movement).".

Looking at [the example app](https://github.com/emilk/egui/blob/master/crates/egui_demo_app/src/backend_panel.rs#L69-L77), it seems like egui will wait for the next input/animation frame unless you explicitly call `ctx.request_repaint()`.

**Comment by psionic12** [comment]
The answer is true for "eframe", not for "egui". "eframe" do has mechanism to check if things changed because it takes the control flow. The question posted is that I found words "reactive mode" in egui codes, so I wondered if there's an efficient way in pure egui to check if nothing new in a new frame.
